utils/deps_xml.py

```diff
@@ -17,35 +17,33 @@
 parser.add_argument(
     "inputfile", type=argparse.FileType('r'), help="Input XML file")
 parser.add_argument(
-    "--file_per_line", action='store_true', help="Output dependencies file per line, rather than Make .d format.")
+    "--file_per_line",
+    action='store_true',
+    help="Output dependencies file per line, rather than Make .d format.")
 
-xi_include = re.compile('<xi:include[^>]*href="([^"]*)"', re.IGNORECASE)
 
-def read_dependencies(f):
-  inputpath = os.path.abspath(f.name)
-  inputdir = os.path.dirname(inputpath)
-  for line in f:
-      line = line.strip()
-      if 'xi:include' not in line:
-          continue
+def read_dependencies(inputfile):
+    inputpath = os.path.abspath(inputfile.name)
+    inputdir = os.path.dirname(inputpath)
+    tree = ET.parse(inputfile)
+    for el in tree.iter():
+        if str(el.tag).endswith('XInclude}include'):
+            yield os.path.abspath(os.path.join(inputdir, el.get('href')))
```

Comment:
Why does this work?  xi:include in the tag visible in the XML?  If this works, can you explain what is going on?

Reply:
The xi is expanded to the URI by the python xml package. so `xi:include` is expanded to `'{http://www.w3.org/2001/XInclude}include'`